File: sdk/java/swim-client/src/main/java/org/vatsim/swim/SwimApiException.java
package org.vatsim.swim;

/**
 * Exception thrown for SWIM API errors
 */
public class SwimApiException extends Exception {
    
    private final int statusCode;
    private final String errorCode;
    
    public SwimApiException(int statusCode, String message) {
        this(statusCode, message, null);
    }
    
    public SwimApiException(int statusCode, String message, String errorCode) {
        super(String.format("[%d] %s: %s", statusCode, errorCode != null ? errorCode : "ERROR", message));
        this.statusCode = statusCode;
        this.errorCode = errorCode;
    }
    
    public int getStatusCode() {
        return statusCode;
    }
    
    public String getErrorCode() {
        return errorCode;
    }
}

File: sdk/php/src/Exceptions/SwimApiException.php
<?php

declare(strict_types=1);

namespace VatSim\Swim\Exceptions;

/**
 * Base exception for SWIM API errors
 */
class SwimApiException extends \Exception
{
    protected int $httpStatusCode;

    public function __construct(string $message = '', int $httpStatusCode = 0, ?\Throwable $previous = null)
    {
        $this->httpStatusCode = $httpStatusCode;
        parent::__construct($message, $httpStatusCode, $previous);
    }

    /**
     * Get HTTP status code
     */
    public function getHttpStatusCode(): int
    {
        return $this->httpStatusCode;
    }
}
